### PR TITLE
Refactor multiple-storage-class-selection tests

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
URL: https://issues.redhat.com/browse/RHSTOR-1873

### CODE REVIEW

**Review by afreen23 — APPROVED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 1]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
...
+    });
+  });
+
   after(() => {
     cy.exec(`echo '${JSON.stringify(testEbsSC)}' | kubectl delete -f -`);
     cy.exec(`echo '${JSON.stringify(testNoProvisionerSC)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume1)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume2)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume3)}' | kubectl delete -f -`);
+    cy.exec('oc get nodes -o json').then((res) => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:71]:
```suggestion
    cy.exec('oc get nodes -l "node-role.kubernetes.io/worker" -o json').then((res) => {
```
Directly get worker nodes.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 2]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
...
+      const nodes = JSON.parse(res.stdout);
+      const allWorkerNodes = nodes.items.filter((node) => getNodeRole(node) === 'worker');
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:73]:
```suggestion
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 3]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
     cy.exec('oc get nodes -o json').then((res) => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:47]:
```suggestion
    cy.exec('oc get nodes -l "node-role.kubernetes.io/worker" -o json').then((res) => {
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 4]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
...
       const nodes = JSON.parse(res.stdout);
       const allWorkerNodes = nodes.items.filter((node) => getNodeRole(node) === 'worker');
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:49]:
```suggestion
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 5]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
...
-    it('New device set is created', () =>
-      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets + 1));
-    it('Device count is 1 in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].count).toBe(1));
-    it('Osd portability is enabled in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].portable).toBe(true));
+
+    cy.log('New device set is created').then(() =>
+      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length + 1),
+    );
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:93]:
```suggestion
    cy.log('Test if new device set is added to the cluster');
    expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length + 1);
```
I don't see the need to chain expect statements to a log statement. Let's change it for all instances.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts [hunk 6]
```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
...
+
+    cy.log('Device count is 1 in the new device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].count).toBe(1),
+    );
+
+    cy.log('Osd portability is disabled in the new device set').then(() =>
+      expect(iAndD.deviceSets[iAndD.index].portable).toBe(false),
+    );
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:147]: This test is used in one other place as well. Perhaps we can create a helper function that tests if new device set is created and device count is added and portability is set as required.(use an args to test if it should be enabled/disabled).

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts (hunk 5 above)
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts:93]: for some reason the `it` block is failing if we are using `expect` without chaining it with `.then` !!
They are working fine with `.then` chaining... 
all I could find was it was written in the function's description that `The expect function is used every time you want to test a value. You will rarely call expect by itself.`

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/multiple-storageclass.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+export const fetchStorageClusterJson = () =>
+  cy.exec(`kubectl get --ignore-not-found storagecluster ${OCS_INTERNAL_CR_NAME} -n ${NS} -o json`);
+
+export const fetchWorkerNodesJson = () =>
+  cy.exec('oc get nodes -l "node-role.kubernetes.io/worker" -o json');
+
+export const addCapacity = (uid: string, scName: string) => {
+  console.log(`UID: ${uid} == SCName: ${scName}`);
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/multiple-storageclass.ts:20]:
```suggestion
```
Debugging?

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/multiple-storageclass.ts:20]: ack... will remove it...
Actually, I did not add this statement... I just migrated the function to the different file...

**Review by bipuladh — APPROVED** [review]
/lgtm

**Pull request merged**